### PR TITLE
Add user customization to `plot_curve` methods

### LINKED ISSUE #986: Allow for additional customization in `pymc_marketing.mmm.plot.plot_curve` function
No description provided.

### PR DESCRIPTION
**Pull request by wd60622**
## Description

User can now specify: 

- same_axes to plot all the curves on the same axes
- axes in the `plot_curve` method
- colors in the `plot_curve` method
- turn legend off and on with `legend` parameter
- Change how the selection is displayed with `sel_to_string` parameter

## Related Issue

- [x] Closes #986 
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1018.org.readthedocs.build/en/1018/

### CODE REVIEW

File: pymc_marketing/mmm/plot.py [hunk 1]
```diff
@@ -235,15 +346,114 @@ def random_samples(
...
+    subplot_kwargs: dict | None = None,
+    axes: npt.NDArray[Axes] | None = None,
+    same_axes: bool = False,
+    colors: Iterable[str] | None = None,
+    legend: bool = False,
+    plot_kwargs: dict[str, Any] | None = None,
+    patch: bool = True,
+    line: bool = True,
+    sel_to_string: SelToString | None = None,
+) -> tuple[plt.Figure, npt.NDArray[Axes]]:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plot.py:369]: I think this auxiliary function would benefit from doc-strings

**Review by juanitorduz — APPROVED** [review]
Nice! I just left some minor comments

File: pymc_marketing/mmm/plot.py (hunk 1 above)
Comment (wd60622) [reply pymc_marketing/mmm/plot.py:322]: Added a bit to it for some context

**Comment by wd60622** [comment]
This should be good to go @juanitorduz

**Pull request merged**